src/AutomationTests/AutomationIntegrationTests.cs

```diff
@@ -94,54 +98,107 @@ public void Scan_Integration_WindowsFormsControlSampler()
             Scan_Integration_Core(WindowsFormsControlSamplerAppPath, WindowsFormsControlSamplerKnownErrorCount);
         }
 
+        [TestMethod]
+        [Timeout(30000)]
+        public void Scan_Integration_WindowsFormsMultiWindowSample()
+        {
+            Scan_Integration_Core(WindowsFormsMultiWindowSamplerAppPath, WindowsFormsMultiWindowSamplerAppAllErrorCount, expectedWindowCount : 2);
+        }
+
+        [TestMethod]
+        [Timeout(30000)]
+        public void Scan_Integration_WindowsFormsMultiWindowSample_SingleWindow()
+        {
+            Scan_Integration_Core(WindowsFormsMultiWindowSamplerAppPath, WindowsFormsMultiWindowSamplerSingleWindowAllErrorCount, false);
+        }
+
         [TestMethod]
         [Timeout(30000)]
         public void Scan_Integration_WpfControlSampler()
         {
             Scan_Integration_Core(WpfControlSamplerAppPath, WpfControlSamplerKnownErrorCount);
         }
 
-        private ScanResults Scan_Integration_Core(string testAppPath, int expectedErrorCount)
+        [TestMethod]
+        [Timeout(30000)]
+        public void SingleWindowScan_MultipleRootsEnabledThrows()
         {
-            LaunchTestApp(testAppPath);
+            LaunchTestApp(WindowsFormsMultiWindowSamplerAppPath);
+            var config = Config.Builder.ForProcessId(TestProcess.Id)
+                .WithOutputDirectory(OutputDir)
+                .WithOutputFileFormat(OutputFileFormat.A11yTest)
+                .WithMultipleScanRootsEnabled()
+                .Build();
+            var scanner = ScannerFactory.CreateScanner(config);
+
+            var action = new Action(() => scanner.Scan());
+            Assert.ThrowsException<InvalidOperationException>(action);
+        }
+
+        [TestMethod]
+        [Timeout(30000)]
+        public void SingleWindowScanWithID_MultipleRootsEnabledThrows()
+        {
+            LaunchTestApp(WindowsFormsMultiWindowSamplerAppPath);
             var config = Config.Builder.ForProcessId(TestProcess.Id)
                 .WithOutputDirectory(OutputDir)
                 .WithOutputFileFormat(OutputFileFormat.A11yTest)
+                .WithMultipleScanRootsEnabled()
                 .Build();
+            var scanner = ScannerFactory.CreateScanner(config);
+
+            var action = new Action(() => scanner.Scan("TestIDForThrow"));
+            Assert.ThrowsException<InvalidOperationException>(action);
+        }
+
+        private ScanResults Scan_Integration_Core(string testAppPath, int expectedErrorCount, bool enableMultipleScanRoots = true, int expectedWindowCount = 1)
```

Comment:
These defaults feel wrong, since the callers each override one value. It seems simpler to make `enableMultipleScanRoots` default to `false`, so that single-root tests override neither value and multi-root tests override both